### PR TITLE
Fail faster Pareto dominance

### PR DESCRIPTION
**Pull request by ow97**
- Assuming CI and reviewers are happy this should supersede esa#392
- Provides anther ~10% improvement on top of esa#392

### CODE REVIEW

**Comment by ow97** [comment]
The Travis CI appears to have failed due to an HTTP error. Based on the assumption that this was a temporary issue I'll trigger a new CI run, by closing and re-opening this PR as per https://stackoverflow.com/a/17624403/4110980.

File: src/utils/multi_objective.cpp [hunk 1]
```diff
@@ -101,16 +101,15 @@ bool pareto_dominance(const vector_double &obj1, const vector_double &obj2)
...
+        } else if (detail::less_than_f(obj1[i], obj2[i])) {
+            foundStrictlyDominatingDimension = true;
         }
-        if (detail::equal_to_f(obj1[i], obj2[i])) {
-            ++count2;
-        }
-    }
-    return (((count1 + count2) == obj1.size()) && (count1 > 0u));
+    } 
+    return foundStrictlyDominatingDimension;
```
Comment (darioizzo) [inline src/utils/multi_objective.cpp]: we typically avoid camel case in pagmo, do you mind changing this variable name to found_strictly_dominating_dimension?

**Comment by darioizzo** [comment]
Thanks for the PR, works for me, will merge when you have implemented the small requested change.

**Comment by ow97** [comment]
Change made as requested. Apologies, I'd missed the note in the wiki on camelCase :)

**Comment by darioizzo** [comment]
its a minimal style realted things, not too important. Thanks for correcting. Merging this.

**Pull request merged**